cdap-common/src/main/java/io/cdap/cdap/common/metadata/QueryParser.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.metadata;
+
+import com.google.common.base.Splitter;
+import io.cdap.cdap.common.metadata.QueryTerm.Qualifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * A thread-safe class that provides helper methods for metadata search string interpretation,
+ * and defines search syntax for qualifying information (e.g. required terms) {@link QueryTerm.Qualifier}.
+ * By default, search items without an operator are considered optional.
+ */
+public class QueryParser {
```

Comment:
Helper class need to be `final` and has a private constructor.

Reply:
Addressed in https://github.com/cdapio/cdap/pull/11478